code/game/gamemodes/gang/gang.dm

```diff
@@ -272,23 +387,58 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
 	return gang_bosses
 
 /datum/game_mode/proc/shuttle_check()
+
 	if(SSshuttle.emergencyNoRecall)
 		return
 	var/alive = 0
 	for(var/mob/living/L in GLOB.player_list)
 		if(L.stat != DEAD)
 			alive++
-
-	if((alive < (GLOB.joined_player_list.len * 0.4)) && ((SSshuttle.emergency.timeLeft(1) > (SSshuttle.emergencyCallTime * 0.4))))
-
+	if((alive <= (GLOB.joined_player_list.len * 0.5)) && (SSshuttle.emergency.mode == SHUTTLE_RECALL || SSshuttle.emergency.mode == SHUTTLE_IDLE || (SSshuttle.emergency.timeLeft(1) > (SSshuttle.emergencyCallTime * 0.4))))
 		SSshuttle.emergencyNoRecall = TRUE
 		SSshuttle.emergency.request(null, set_coefficient = 0.4)
 		priority_announce("Catastrophic casualties detected: crisis shuttle protocols activated - jamming recall signals across all frequencies.")
+	if(vigilantes)
+		posse_timer++
+		if((alive < (GLOB.joined_player_list.len *  0.75)) && posse_timer >= 4)
+			posse_timer = 0
+			vigilante_vengeance()
+
+
+/datum/game_mode/proc/vigilante_vengeance()
+	var/list/mob/dead/observer/candidates = pollGhostCandidates("Would you like be a part of a Vigilante posse?", "pAI", null, FALSE, 100)
```

Comment:
I dunno I saw this used frequently for other modes so I figured pAI ban = ghost role ban